spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -1,77 +1,127 @@
 require "rails_helper"
 
-# TODO: [@thepracticaldev/delightful] Reuse this shared example across all notifications,
-# since it should be tested against every kind of broadcast we could send.
-RSpec.shared_examples "unsubscribed from welcome notifications" do |_broadcast|
-  it "does not send a notification to an unsubscribed user" do
-    expect do
-      sidekiq_perform_enqueued_jobs { described_class.call(unsubscribed_user.id) }
-    end.to not_change(unsubscribed_user.notifications, :count)
+RSpec.describe Broadcasts::WelcomeNotification::Generator, type: :service do
+  let(:mascot_account)             { create(:user) }
+  let!(:welcome_thread)            { create(:article, user: mascot_account, published: true, tags: "welcome") }
+  let!(:welcome_broadcast)         { create(:welcome_broadcast) }
+  let!(:twitter_connect_broadcast) { create(:twitter_connect_broadcast) }
+  let!(:github_connect_broadcast)  { create(:github_connect_broadcast) }
+
+  before do
+    allow(User).to receive(:mascot_account).and_return(mascot_account)
+    SiteConfig.staff_user_id = mascot_account.id
+  end
+
+  after do
+    # SiteConfig.clear_cache should work here but for some reason it isn't
+    SiteConfig.staff_user_id = 1
+  end
+
+  it "requires a valid user id" do
+    expect { described_class.call(1) }.to raise_error(ActiveRecord::RecordNotFound)
   end
-end
 
-RSpec.describe Broadcasts::WelcomeNotification::Generator, type: :service do
   describe "::call" do
-    let(:user) { create(:user) }
-    let(:unsubscribed_user) { create(:user, welcome_notifications: false) }
-    let(:mascot_account) { create(:user) }
-    let!(:welcome_broadcast) { create(:welcome_broadcast, :active) }
+    it "send all forms of welcome notifications" do
+      user = create(:user, :with_identity, identities: ["github"], created_at: 1.week.ago)
+      sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
+      expect(user.notifications.count).to eq(2)
```

Comment:
why should the user receive `2` notifications?

Reply:
oh wait...is this because they should have received the welcome thread notification? if so, i think this test doesn't read very clearly to me. if it's just meant to test whether a user can receive any notification, i'd expect that the count of notifications wouldn't matter

Reply:
> is it possible right now for both send_welcome_notification and send_authentication_notification to run?
actually this is doing exactly that.